src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -150,7 +151,7 @@ public static void ValidateReduceResultsCollectionName(IndexDefinition definitio
                 }
             }
 
-            if (isIndexReset == false)
+            if (validateMapReduceCollectionName)
             {
```

Comment:
If we will check here that we have index with same name and same output collection then we can skip that 'if'? Wouldn't that be the case?

If we previously allowed a creation of index 'MyIndex' that Outputs to collection 'Companies' then if we have still that index and we are overriding it then the assumption that it was valid configuration still takes place?

Reply:
https://github.com/ppekrol/ravendb/commit/22e90f9c337c58a3ca6e42513d2128b729656497 seems to work?

Reply:
You are still creating an errored index.
(I don't think that there is a test for that)
1) deploy a map-reduce index with output to collection with some documents
2) modify the index definition and deploy.
There will be an errored index (side by side one in this case) while it shouldn't been even deployed.

Reply:
Correct.

We need a test for this case.